rasa/nlu/utils/__init__.py

```diff
@@ -37,11 +35,16 @@ def is_model_dir(model_dir: Text) -> bool:
     specifically checks if the directory has no subdirectories and
     if all files have an appropriate ending."""
     allowed_extensions = {".json", ".pkl", ".dat"}
-    dir_tree = list(os.walk(model_dir))
-    if len(dir_tree) != 1:
+
+    dir_tree = Path(model_dir)
+    if not dir_tree.is_dir():
         return False
-    model_dir, child_dirs, files = dir_tree[0]
-    file_extenstions = [os.path.splitext(f)[1] for f in files]
+
+    iter_dir = [d for d in dir_tree.iterdir()]
```

Comment:
Could we replace lines 43 and 44 with a single line `any()` statement? I think that still remains readable.